migrations/2019-03-30-212058_create_post_view/up.sql

```diff
@@ -1,51 +1,107 @@
 -- Rank = ScaleFactor * sign(Score) * log(1 + abs(Score)) / (Time + 2)^Gravity
-create or replace function hot_rank(
-  score numeric,
-  published timestamp without time zone)
-returns integer as $$
-begin
-  -- hours_diff:=EXTRACT(EPOCH FROM (timezone('utc',now()) - published))/3600
-  return floor(10000*log(greatest(1,score+3)) / power(((EXTRACT(EPOCH FROM (timezone('utc',now()) - published))/3600) + 2), 1.8))::integer;
-end; $$
+CREATE OR REPLACE FUNCTION hot_rank (score numeric, published timestamp without time zone)
+    RETURNS integer
+    AS $$
+BEGIN
+    -- hours_diff:=EXTRACT(EPOCH FROM (timezone('utc',now()) - published))/3600
+    RETURN floor(10000 * log(greatest (1, score + 3)) / power(((EXTRACT(EPOCH FROM (timezone('utc', now()) - published)) / 3600) + 2), 1.8))::integer;
+END;
+$$
 LANGUAGE plpgsql;
 
-create view post_view as
-with all_post as
-(
-  select        
-  p.*,
-  (select name from user_ where p.creator_id = user_.id) as creator_name,
-  (select name from community where p.community_id = community.id) as community_name,
-  (select removed from community c where p.community_id = c.id) as community_removed,
-  (select count(*) from comment where comment.post_id = p.id) as number_of_comments,
-  coalesce(sum(pl.score), 0) as score,
-  count (case when pl.score = 1 then 1 else null end) as upvotes,
-  count (case when pl.score = -1 then 1 else null end) as downvotes,
-  hot_rank(coalesce(sum(pl.score) , 0), p.published) as hot_rank
-  from post p
-  left join post_like pl on p.id = pl.post_id
-  group by p.id
+CREATE VIEW post_view AS
+with all_post AS (
+    SELECT
+        p.*,
+        (
+            SELECT
+                name
+            FROM
+                user_
+            WHERE
+                p.creator_id = user_.id) AS creator_name,
+        (
+            SELECT
+                name
+            FROM
+                community
+            WHERE
+                p.community_id = community.id) AS community_name,
+        (
+            SELECT
+                removed
+            FROM
+                community c
+            WHERE
+                p.community_id = c.id) AS community_removed,
+        (
+            SELECT
+                count(*)
+            FROM
+                comment
+            WHERE
+                comment.post_id = p.id) AS number_of_comments,
```

Comment:
However this kind of formatting isnt good at all. Makes queries almost unreadable with every word on a separate line. Anyway this can be changed?

Reply:
It might be able to be tweaked, but I do find it readable. More than anything I'd like to keep the defaults that `pg_format` uses, just like we don't tweak prettier or any other formatter.